modules/git/repo_tag.go

```diff
@@ -35,34 +36,66 @@ func (repo *Repository) CreateTag(name, revision string) error {
 	return err
 }
 
+// CreateAnnotatedTag create one annotated tag in the repository
+func (repo *Repository) CreateAnnotatedTag(name, message, revision string) error {
+	_, err := NewCommand("tag", "-a", "-m", message, name, revision).RunInDir(repo.Path)
+	return err
+}
+
 func (repo *Repository) getTag(id SHA1) (*Tag, error) {
 	t, ok := repo.tagCache.Get(id.String())
 	if ok {
 		log("Hit cache: %s", id)
-		return t.(*Tag), nil
+		tagClone := *t.(*Tag)
```

Comment:
NOTE: This clones a struct, which we need to do here from cache since multiple tags can have the same SHA1 (the tag name gets set/changed in the function that called getTag). Otherwise 2 or more tags with the same commit will all have the same tag name when the JSON is returned.